dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1002,22 +1013,18 @@ void TeachingTip::OnHeroContentPlacementChanged()
     switch (HeroContentPlacement())
     {
     case winrt::TeachingTipHeroContentPlacementMode::Auto:
-        break;
-    case winrt::TeachingTipHeroContentPlacementMode::Top:
-        winrt::VisualStateManager::GoToState(*this, L"HeroContentTop"sv, false);
-        if (m_currentHeroContentEffectivePlacementMode != winrt::TeachingTipHeroContentPlacementMode::Top)
+        // If there is no target for the TeachingTip we don't display a tail. As such, we will always display the hero content
+        // in the header of the TeachingTip.
+        if (!Target())
         {
-            m_currentHeroContentEffectivePlacementMode = winrt::TeachingTipHeroContentPlacementMode::Top;
-            TeachingTipTestHooks::NotifyEffectiveHeroContentPlacementChanged(*this);
+            DoUpdateDynamicHeroContentPlacementToTop();
         }
```

Comment:
I believe the no-opt was intentional, in the Auto case we do not want to reposition the hero content while the tip is open, so delay processing this change until the next time we open.

Reply:
That will lead to an issue where the VisualStateGroup "HeroContentPlacementStates" has no current state when the tip is launched in the Auto HeroContentPlacement mode. And according to the documentation, this should not happen: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.visualstategroup.currentstate?view=winrt-19041#remarks

Since we place the HeroContent at the top of an untargeted TeachingTip given a HeroContentPlacement of Auto we should put the VisualStateGroup in the HeroContentTop state here which is what DoUpdateDynamicHeroContentPlacementToTop() will do.

If an update of the HeroContentPlacement setting to Auto is still not desired while an *untargeted* TeachingTip is open I can add an additional check here to only call DoUpdateDynamicHeroContentPlacementToTop() when the tip is untargeted *and* unopened. Though I'm not sure why we would want to block that for untargeted TeachingTips, since in their case, the hero content will always be placed at the top of the tip and the tip itself does not have a tail we have to take into consideration here.

Reply:
the Auto scenario means the app author wants us to decide was is best, if they wanted the hero content to be top then they could set the the property to top.  We no-opt here because we believe that switching the hero content position from bottom to top while the tip is open is usually a jarring experience so we delay the switch until the next opening.  Why is the null state an issue for untargetted but not targetted tips. Can we change the the Untargetted opening logic to do the same thing the targetted one does with respects to actively choosing a hero content state?


---
In reply to: [487262633](https://github.com/microsoft/microsoft-ui-xaml/pull/3271#discussion_r487262633) [](ancestors = 487262633)

Reply:
Reverted this change (removed that call). An untargetd TeachignTip will now behave the same as a targeted TeachingTip regarding its HeroContentPlacement visual state group: Once the tip is open either the Top or Bottom visual state wil be entered. Before the tip has been opened, no state is applied (HeroContentPlacementStates.CurrentState = null).